CondFormats/EcalObjects/interface/EcalRecHitParametersSoA.h

```diff
@@ -0,0 +1,21 @@
+#ifndef CondFormats_EcalObjects_EcalRecHitParametersSoA_h
+#define CondFormats_EcalObjects_EcalRecHitParametersSoA_h
+
+#include <bitset>
+#include <array>
+#include "DataFormats/EcalRecHit/interface/EcalRecHit.h"
+#include "DataFormats/SoATemplate/interface/SoACommon.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+#include "DataFormats/SoATemplate/interface/SoAView.h"
+
+constexpr size_t kNEcalChannelStatusCodes = 16;  // The HW supports 16 channel status codes
+using RecoFlagBitsArray =
+    std::array<uint32_t, kNEcalChannelStatusCodes>;  // associate recoFlagBits to all channel status codes
+
+GENERATE_SOA_LAYOUT(EcalRecHitParametersSoALayout,
+                    SOA_SCALAR(RecoFlagBitsArray, recoFlagBits),
+                    SOA_SCALAR(std::bitset<kNEcalChannelStatusCodes>, channelStatusCodesToBeExcluded))
+
+using EcalRecHitParametersSoA = EcalRecHitParametersSoALayout<>;
```

Comment:
Just thinking, could this be more simply
```c++
struct EcalRecHitParameters {
    RecoFlagBitsArray recoFlagBits;  
    std::bitset<kNEcalChannelStatusCodes> channelStatusCodesToBeExcluded;
};
```

and then use a `PortableObject`:
```c++
using EcalRecHitParametersHost = PortableHostObject<EcalRecHitParameters>;
```

_etc_ ?

Reply:
I thought about this as well but in that case there would not be automatic host to device transfers of the parameters if I understand things correctly.

Reply:
Actually, `PortableHostObject<T>` should have all the same automatism as `PortableHostCollection<T>`.

If it doesn't work we should fix it.

Reply:
In that case a `PortableHostObject` would be simpler to use.